modules/components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/eventsimulator/core/simulator/csvFeedSimulation/core/CSVFeedEventSimulator.java

```diff
@@ -0,0 +1,266 @@
+        if (delay <= 0) {
+            log.warn("Events will be sent continuously since the delay between events are set to "
+                    + delay + " milliseconds");
+            delay = 0;
+        }
+
+        LinkedHashMap<String,StreamDefinitionRetriever.Type> streamDefinition =
+                EventSimulatorDataHolder.getInstance().getStreamDefinitionService().streamDefinitionService(csvFileConfig.getStreamName());
+
+        try {
+            // Initialize Reader
+            if (FileStore.getFileStore().checkExists(csvFileConfig.getFileName())) {
+                in = new FileReader(String.valueOf(Paths.get(System.getProperty("java.io.tmpdir"), csvFileConfig.getFileDto().getFileInfo().getFileName())));
+
+                // Initialize CSVParser with appropriate CSVFormat according to delimiter
+                if (!csvFileConfig.getDelimiter().isEmpty()) {
+                    switch (csvFileConfig.getDelimiter()) {
+                        case ",":
+                            csvParser = new CSVParser(in, CSVFormat.DEFAULT);
+                            break;
+                        case ";":
+                            csvParser = new CSVParser(in, CSVFormat.EXCEL);
+                            break;
+                        case "\\t":
+                            csvParser = new CSVParser(in, CSVFormat.TDF);
+                            break;
+                        default:
+                            csvParser = new CSVParser(in, CSVFormat.newFormat(csvFileConfig.getDelimiter().charAt(0)));
+                    }
+
+                    int attributeSize = streamDefinition.size();
+                    Long timestamp;
+
+//            create a treemap to hold csv file data. the key would be the timestamp and value will be a list of events
+                    TreeMap<Long, ArrayList<Event>> eventsMap = new TreeMap<>();
+
+                    // Iterate through the CSV file line by line
+                    for (CSVRecord record : csvParser) {
+                        try {
+                            if (!isPaused) {
+                                if (record.size() != attributeSize) {
+                                    log.warn("No of attribute is not equal to attribute size: " + attributeSize + " is needed" + "in Row no:" + noOfEvents + 1);
+                                }
+                                String[] attributes = new String[attributeSize];
+                                noOfEvents = csvParser.getCurrentLineNumber();
+
+                                for (int i = 0; i < record.size(); i++) {
+                                    attributes[i] = record.get(i);
+                                }
+
+                                //convert Attribute values into event
+                                Event event = EventConverter.eventConverter(streamDefinition, attributes);
+                                System.out.println("Input Event (CSV feed)" + Arrays.deepToString(event.getData()));
+
```

Comment:
Is it required to use a System.out.println here?
Can't we use a printstream here ? Ideally wso2 check styles should catch System.out.println statements, haven't we run check styles?